Apps/WebClient/src/Server/Controllers/BetaRequestController.cs

```diff
@@ -0,0 +1,139 @@
+    using HealthGateway.WebClient.Services;
+    using Microsoft.AspNetCore.Authorization;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+
+    /// <summary>
+    /// Web API to handle requests for beta users.
+    /// </summary>
+    [Authorize]
+    [ApiVersion("1.0")]
+    [Route("v{version:apiVersion}/api/[controller]")]
+    [ApiController]
+    public class BetaRequestController
+    {
+        private readonly IBetaRequestService betaRequestService;
+        private readonly IHttpContextAccessor httpContextAccessor;
+        private readonly IAuthorizationService authorizationService;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BetaRequestController"/> class.
+        /// </summary>
+        /// <param name="betaRequestService">The injected beta request service.</param>
+        /// <param name="httpContextAccessor">The injected http context accessor provider.</param>
+        /// <param name="authorizationService">The injected authorization service.</param>        
+        public BetaRequestController(
+            IBetaRequestService betaRequestService,
+            IHttpContextAccessor httpContextAccessor,
+            IAuthorizationService authorizationService)
+        {
+            this.httpContextAccessor = httpContextAccessor;
+            this.authorizationService = authorizationService;
+            this.betaRequestService = betaRequestService;
+        }
+
+        /// <summary>
+        /// Posts a beta request json to be inserted into the database.
+        /// </summary>
+        /// <returns>The http status.</returns>
+        /// <param name="hdid">The resource hdid.</param>
+        /// <param name="betaRequest">The beta request model.</param>
+        /// <response code="200">The beta request record was saved.</response>
+        /// <response code="400">The beta request was already inserted.</response>
+        /// <response code="401">The client must authenticate itself to get the requested response.</response>
+        /// <response code="403">The client does not have access rights to the content; that is, it is unauthorized, so the server is refusing to give the requested resource. Unlike 401, the client's identity is known to the server.</response>
+        [HttpPost]
+        [Route("{hdid}")]
+        [Authorize(Policy = "PatientOnly")]
+        public async Task<IActionResult> CreateBetaRequest(string hdid, [FromBody] BetaRequest betaRequest)
+        {
+            Contract.Requires(hdid != null);
+            Contract.Requires(betaRequest != null);
+
+            // Validate that the query parameter matches the post body
+            if (!hdid.Equals(betaRequest.hdid, StringComparison.CurrentCultureIgnoreCase))
+            {
+                return new BadRequestResult();
+            }
+
+            // Validate the hdid to be a patient.
+            ClaimsPrincipal user = this.httpContextAccessor.HttpContext.User;
```

Comment:
We're doing a lot of this now, should we think about creating a technical task to refactor to utility code?

Reply:
100% agree. I've created an unparented task to address this